include/dnnl.hpp

```diff
@@ -37,6 +37,38 @@
 #endif
 /// @endcond
 
+// __cpp_exceptions is referred from
+// https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_exceptions.html
+// gcc < 5 does not define __cpp_exceptions but __EXCEPTIONS, 
+// _CPPUNWIND is used by MSVC, see
+// https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019
+#ifndef DNNL_ENABLE_EXCEPTIONS
+ #if __cpp_exceptions || __EXCEPTIONS || \
+     (defined(_MSC_VER) && defined(_CPPUNWIND))
+#define DNNL_ENABLE_EXCEPTIONS 1
+#else
+#define DNNL_ENABLE_EXCEPTIONS 0
+#endif
+#endif
+
+#if defined(__GNUC__) || defined(__clang__)
+#define dnnl_trap() __builtin_trap()
+#elif defined(__INTEL_COMPILER) || defined(_MSC_VER)
+#define dnnl_trap() __debugbreak()
+#else
+  #error "unknown compiler"
+#endif
+
+#if DNNL_ENABLE_EXCEPTIONS
+#define DNNL_THROW(status, msg) throw error(status, msg);
+#else
+#define DNNL_THROW(status, msg) \
+    do {                        \
+        fprintf(stderr, msg);   \
+        dnnl_trap();            \
+    } while (0);
+#endif
+
```

Comment:
This code requires formatting with clang-format according to product guide-lines.

Reply:
done